### ISSUE TITLE
Bugs in Select.withSearch()

### ISSUE THREAD

**Issue by patricknicolosi** [nank1ro/flutter-shadcn-ui#116]
### Steps to reproduce

follow the code below

### Expected results

- Search widget in ShadSelect.withSearch responsiveness
- No issues during a search

### Actual results

- Search Widget out of screen
- Range error during a search

### shadcn_ui version
From pubspec.yaml

```
shadcn_ui :
  git:
    url: https://github.com/nank1ro/flutter-shadcn-ui
```
### Platform

Windows

### Code sample

cities.json has content of [comuni.json](https://raw.githubusercontent.com/matteocontrini/comuni-json/master/comuni.json)
```dart
import 'dart:convert';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:gym_manager/models/city.dart';
import 'package:shadcn_ui/shadcn_ui.dart';

class CityPicker extends StatefulWidget {
  final ValueChanged<City?> onChanged;
  final City? initialValue;

  const CityPicker(
      {required this.onChanged, required this.initialValue, super.key});

  @override
  State<CityPicker> createState() => _CityPickerState();
}

class _CityPickerState extends State<CityPicker> {
  String? _searchValue;
  City? _selectedCity;
  late Future<List<City>> _readCitiesFromAsset;

  @override
  void initState() {
    _selectedCity = widget.initialValue;
    _readCitiesFromAsset = _readFromAsset();
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return FutureBuilder<List<City>>(
      future: _readCitiesFromAsset,
      builder: (context, snapshot) {
        if (snapshot.connectionState == ConnectionState.waiting) {
          return Padding(
            padding: const EdgeInsets.all(50.0),
            child: const ShadProgress(),
          );
        } else {
          List<City> cities = snapshot.data
                  ?.where((e) =>
                      e.name
                          ?.toLowerCase().trim()
                          .contains(_searchValue?.toLowerCase().trim() ?? "") ??
                      false)
                  .toList() ??
              [];
          return Padding(
            padding: const EdgeInsets.only(top: 12, bottom: 20),
            child: ShadSelect<City>.withSearch(
              initialValue: _selectedCity,
              placeholder: const Text('Seleziona città'),
              onSearchChanged: (value) => setState(() {
                _searchValue = value;
              }),
              onChanged: (value) => setState(() {
                widget.onChanged(value);
                _selectedCity = value;
              }),
              searchPlaceholder: const Text('Cerca città'),
              optionsBuilder: (context, index) => ShadOption(
                value: cities[index],
                child: Text(cities[index].name ?? ""),
              ),
              selectedOptionBuilder: (context, value) =>
                  Text(_selectedCity?.name ?? ""),
            ),
          );
        }
      },
    );
  }

  Future<List<City>> _readFromAsset() async {
    String citiesString = await rootBundle.loadString("assets/cities.json");
    List jsonCities = jsonDecode(citiesString) as List;
    List<City> cities = jsonCities.map((city) => City.fromJson(city)).toList();
    return cities;
  }
}
```

### Search widget out of screen
<img width="377" alt="Immagine 2024-07-19 201820" src="https://github.com/user-attachments/assets/57fcf407-7819-4030-b091-ea224c0bb847">

### Range issue during search
<img width="379" alt="Immagine 2024-07-31 115031" src="https://github.com/user-attachments/assets/56890a3f-6c5d-4779-9189-f2250c39801f">

### Flutter Doctor output

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.2, on Microsoft Windows [Versione 10.0.22631.3880], locale it-IT)
[✓] Windows Version (Installed version of Windows is version 10 or higher)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✓] Chrome - develop for the web
[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.4)
[!] Android Studio (not installed)
[✓] VS Code (version 1.91.1)
[✓] Connected device (3 available)
[✓] Network resources

**Comment by patricknicolosi** [comment]
In the latest version the maxWidth parameter is no longer there

> Check the following code, the keypoints are `maxWidth: 400` and `return null` inside the `optionsBuilder` to end building items.
> 
> ```dart
> ShadSelect<City>.withSearch(
>                 maxWidth: 400,
>                 initialValue: _selectedCity,
>                 placeholder: const Text('Seleziona città'),
>                 onSearchChanged: (value) => setState(() {
>                   _searchValue = value;
>                 }),
>                 onChanged: (value) => setState(() {
>                   widget.onChanged(value);
>                   _selectedCity = value;
>                 }),
>                 searchPlaceholder: const Text('Cerca città'),
>                 optionsBuilder: (context, index) {
>                   // no more cities to render
>                   if (index >= cities.length) {
>                     return null;
>                   }
>                   return ShadOption(
>                     value: cities[index],
>                     child: Text(cities[index].name ?? ""),
>                   );
>                 },
>                 selectedOptionBuilder: (context, value) =>
>                     Text(_selectedCity?.name ?? ""),
>               )
> ```

**Comment by nank1ro** [comment]
Are you sure?
Which version are you using? I haven't removed it

**Comment by patricknicolosi** [comment]
> Are you sure? Which version are you using? I haven't removed it

From my pubspec shadcn_ui: ^0.7.3

**Comment by patricknicolosi** [comment]
> Can you try removing the `pubspec.lock` and do `flutter pub get` again? I can see and use the `maxWidth` parameter

Sorry, the problem occurs with ShadSelectFormField.withSearch and no with ShadSelect<City>.withSearch

**Comment by nank1ro** [comment]
> > Can you try removing the `pubspec.lock` and do `flutter pub get` again? I can see and use the `maxWidth` parameter
> 
> Sorry, the problem occurs with ShadSelectFormField.withSearch and no with ShadSelect.withSearch

Yes you're right, I forgot about it. Can you point to the latest version on Github for now?
I will release a new version on pub.dev in the next days with the fix

**Comment by nank1ro** [comment]
closing cause I just released the v0.8.0
